src/python/pants/util/retry.py

```diff
@@ -30,4 +30,5 @@ def retry_on_exception(func, max_retries, exception_types, backoff_func=lambda n
       return func()
     except exception_types as e:
       logger.debug('encountered exception on retry #{}: {!r}'.format(i, e))
-  raise
+      if i == max_retries - 1:
+        raise e
```

Comment:
Should this be `raise e` or just `raise`? My understanding was the latter preserves the original stacktrace, which we would want here.

Reply:
You're right! Thanks, fixed with newest commit.